src/commands/generateFunctions.ts

```diff
@@ -0,0 +1,241 @@
+const identifierStartCharacter: string = `(${letterCharacter}|_)`;
+const identifierOrKeyword: string = `${identifierStartCharacter}(${identifierPartCharacter})*`;
+// tslint:disable-next-line: no-unsafe-any
+const identifierRegex: RegExp = XRegExp(`^${identifierOrKeyword}$`);
+// Keywords: https://github.com/dotnet/csharplang/blob/master/spec/lexical-structure.md#keywords
+const keywords: string[] = ['abstract', 'as', 'base', 'bool', 'break', 'byte', 'case', 'catch', 'char', 'checked', 'class', 'const', 'continue', 'decimal', 'default', 'delegate', 'do', 'double', 'else', 'enum', 'event', 'explicit', 'extern', 'false', 'finally', 'fixed', 'float', 'for', 'foreach', 'goto', 'if', 'implicit', 'in', 'int', 'interface', 'internal', 'is', 'lock', 'long', 'namespace', 'new', 'null', 'object', 'operator', 'out', 'override', 'params', 'private', 'protected', 'public', 'readonly', 'ref', 'return', 'sbyte', 'sealed', 'short', 'sizeof', 'stackalloc', 'static', 'string', 'struct', 'switch', 'this', 'throw', 'true', 'try', 'typeof', 'uint', 'ulong', 'unchecked', 'unsafe', 'ushort', 'using', 'virtual', 'void', 'volatile', 'while'];
+
+// tslint:disable: no-non-null-assertion
+export async function generateFunctions(node?: ApiTreeItem): Promise<void> {
+    if (!node) {
+        node = <ApiTreeItem>await ext.tree.showTreeItemPicker(ApiTreeItem.contextValue);
+    }
+
+    ext.outputChannel.show();
+
+    const openApiEditor: OpenApiEditor = new OpenApiEditor();
+    const openAPIDocString = await openApiEditor.getData(node);
+
+    if (!await validateAutorestInstalled()) {
+        return;
+    }
+
+    // tslint:disable-next-line: no-unsafe-any
+    const languages : string[] = Object.keys(languageTypes).map(key => languageTypes[key]);
+    const language = await ext.ui.showQuickPick(
+        languages.map((s) => {return { label: s, description: '', detail: '' }; }), { placeHolder: "Select language", canPickMany: false});
+
+    if (!await checkEnvironmentInstalled(language.label)) {
+        throw new Error(`You haven't installed '${language.label}' on your machine, please install '${language.label}' to continue.`);
+    }
+
+    let namespace = "";
+    if (language.label === languageTypes.Java) {
+        namespace = await askJavaNamespace();
+    } else if (language.label === languageTypes.CSharp) {
+        namespace = await askCSharpNamespace();
+    }
+
+    const uris = await askFolder();
+
+    const openAPIFilePath = path.join(uris[0].fsPath, `${node!.apiContract.name}.json`);
+
+    await window.withProgress(
+        {
+            location: ProgressLocation.Notification,
+            title: localize("openAPI", `Downloading OpenAPI Document for API '${node.apiContract.name}'...`),
+            cancellable: false
+        },
+        async () => {
+            await fse.writeFile(openAPIFilePath, openAPIDocString);
```

Comment:
Do we only download v3? or we give that choice?

Reply:
we don't have choices, since we only support get data openAPI 3.0